### PR TITLE
Update common.h header and reduce struct registers usage

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/common.h [hunk 1]
```diff
@@ -23,17 +23,16 @@
...
 #endif
 
-#include <assert.h>
-#include <math.h>
-#include <stddef.h>
-#include <stdint.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <stdbool.h>
```
Comment (janisozaur) [inline src/openrct2/common.h]: Note `cstdbool` include is gone, as it is a) not required b) deprecated in C++17

File: src/openrct2/paint/tile_element/Surface.cpp [hunk 2]
```diff
@@ -604,29 +604,29 @@ static void viewport_surface_draw_tile_side_bottom(paint_session * session, enum
...
+        ah = 1;
     }
     else
     {
         if (isWater)
         {
-            regs.dh = map_get_water_height(neighbour.tile_element);
-            if (regs.dl == regs.dh)
+            dh = map_get_water_height(neighbour.tile_element);
+            if (dl == dh)
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Surface.cpp]: AppVeyor chokes on the possibility uninitialised usage of `dl`. Should perhaps default to 0.

Comment (janisozaur) [reply src/openrct2/paint/tile_element/Surface.cpp]: done

File: src/openrct2/paint/tile_element/Surface.cpp [hunk 3]
```diff
@@ -559,7 +559,7 @@ static void viewport_surface_draw_tile_side_bottom(paint_session * session, enum
     if (!is_csg_loaded() && edgeStyle >= TERRAIN_EDGE_RCT2_COUNT)
         edgeStyle = TERRAIN_EDGE_ROCK;
 
-    registers regs{};
+    sint16 al, ah, cl, ch, dl = 0, dh;
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/Surface.cpp]: All of the variables in this function can be named. No point wasting time refactoring to another interim variable name.

File: src/openrct2/paint/tile_element/Surface.cpp [hunk 4]
```diff
@@ -770,30 +770,30 @@ static void viewport_surface_draw_tile_side_top(paint_session * session, enum ed
     if (!is_csg_loaded() && terrain >= TERRAIN_EDGE_RCT2_COUNT)
         terrain = TERRAIN_EDGE_ROCK;
 
-    registers regs{};
+    sint16 al, ah, cl, ch, dl = 0, dh;
```
Comment (duncanspumpkin) [inline src/openrct2/paint/tile_element/Surface.cpp]: As above

File: src/openrct2/ride/Ride.cpp [hunk 5]
```diff
@@ -4530,11 +4529,11 @@ static rct_vehicle *vehicle_create_car(
     vehicle->vehicle_type = vehicleEntryIndex;
     vehicle->is_child = carIndex == 0 ? 0 : 1;
     vehicle->var_44 = ror32(vehicleEntry->spacing, 10) & 0xFFFF;
-    regs.edx = vehicleEntry->spacing >> 1;
-    *remainingDistance -= regs.edx;
+    edx = vehicleEntry->spacing >> 1;
+    *remainingDistance -= edx;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Ride.cpp]: Use edxs value directly and remove the variable

**Pull request merged**